File: src/main/java/no/nav/fo/forenkletdeploy/unleash/streategies/ByEnvironmentStrategy.java
package no.nav.fo.forenkletdeploy.unleash.streategies;

import no.finn.unleash.strategy.Strategy;
import no.nav.fo.forenkletdeploy.service.UnleashService;

import java.util.Arrays;
import java.util.List;
import java.util.Map;

public class ByEnvironmentStrategy implements Strategy {
    @Override
    public String getName() {
        return "byEnvironment";
    }

    @Override
    public boolean isEnabled(Map<String, String> parameters) {
        String env = UnleashService.getEnvironment();
        List<String> enabledEnvs =  Arrays.asList(parameters.getOrDefault("miljø", "").split(","));
        return enabledEnvs.contains(env);
    }
}

File: src/main/kotlin/no/nav/fo/forenkletdeploy/unleash/streategies/ByEnvironmentStrategy.kt
package no.nav.fo.forenkletdeploy.unleash.streategies

import no.finn.unleash.strategy.Strategy
import no.nav.fo.forenkletdeploy.util.Utils.getEnvironment

import java.util.Arrays

class ByEnvironmentStrategy : Strategy {
    override fun getName(): String {
        return "byEnvironment"
    }

    override fun isEnabled(parameters: Map<String, String>?): Boolean {
        val enabledEnvs = parameters?.getOrDefault("miljø", "")?.split(",") ?: emptyList()
        return enabledEnvs.contains(getEnvironment())
    }
}
